src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.cs

```diff
@@ -244,14 +246,25 @@ protected internal override void SetPinMode(int pinNumber, PinMode mode)
                     _ => -1,
                 };
 
+                if (requestResult == -1)
+                {
+                    throw ExceptionHelper.GetIOException(ExceptionResource.SetPinModeError, Marshal.GetLastWin32Error(),
+                        pinNumber);
+                }
+
                 pinHandle.PinMode = mode;
+                return;
             }
 
-            if (requestResult == -1)
-            {
-                throw ExceptionHelper.GetIOException(ExceptionResource.SetPinModeError, Marshal.GetLastWin32Error(),
-                    pinNumber);
-            }
+            throw new InvalidOperationException($"Pin {pinNumber} is not open");
+        }
+
+        /// <inheritdoc />
+        protected internal override void SetPinMode(int pinNumber, PinMode mode, PinValue initialValue)
+        {
+            // On the Raspberry Pi, we can Write the out value even if the mode is something other than out. It will take effect once we change the mode
```

Comment:
This comment says that this behavior is specific to Raspberry Pi, but does this apply as well to other boards? Libgpiod driver could be used in other boards so we should double check to make sure that other boards won't break with this.

Reply:
Well, it works on the Raspberry Pi (both with the Raspberry Pi driver and the Libgpiod driver), whether other boards would fail on this I cannot tell. At least the libgpiod code does not include any test that could throw here, except standard error handling on an Ioctl call.